modules/collator-selection/src/lib.rs

```diff
@@ -406,30 +397,83 @@ pub mod pallet {
 	/// Play the role of the session manager.
 	impl<T: Config> SessionManager<T::AccountId> for Pallet<T> {
 		fn new_session(index: SessionIndex) -> Option<Vec<T::AccountId>> {
-			log::info!(
-				"assembling new collators for new session {} at #{:?}",
+			let candidates = Self::candidates()
+				.iter()
+				.map(|candidate| candidate.who.clone())
+				.collect::<Vec<_>>();
+			let result = Self::assemble_collators(candidates);
+
+			log::debug!(
+				"assembling new collators for new session {:?} at #{:?}, candidates: {:?}",
 				index,
 				<frame_system::Pallet<T>>::block_number(),
+				result,
 			);
 
-			let candidates = Self::candidates();
-			let candidates_len_before = candidates.len();
-			let active_candidates = Self::kick_stale_candidates(candidates.into_inner());
-			let active_candidates_len = active_candidates.len();
-			let result = Self::assemble_collators(active_candidates);
-			let removed = candidates_len_before - active_candidates_len;
+			//TODO
+			//frame_system::Pallet::<T>::register_extra_weight_unchecked(
+			//	T::WeightInfo::new_session(candidates_len_before as u32, removed as u32),
+			//	DispatchClass::Mandatory,
+			//);
 
-			frame_system::Pallet::<T>::register_extra_weight_unchecked(
-				T::WeightInfo::new_session(candidates_len_before as u32, removed as u32),
-				DispatchClass::Mandatory,
-			);
 			Some(result)
 		}
-		fn start_session(_: SessionIndex) {
-			// we don't care.
+
+		fn start_session(index: SessionIndex) {
+			<Candidates<T>>::mutate(|candidates| {
+				let mut candidates_inner = candidates.to_vec();
+				candidates_inner.iter_mut().for_each(|candidate| {
+					if candidate.validator {
+						<SessionPoints<T>>::insert(&candidate.who, 0);
```

Comment:
I think we need to do this for all new authors. otherwise in `end_session`, one that never produce block will not in `SessionPoints`

Reply:
If new authors in `SessionPoints`, it will be removed in `end_session`.
Here delay one session.

Reply:
we know on which session index the candidate will become author, so we can insert all new authors